src/Api/PubnubApi/Transport/HttpClientService.cs

```diff
@@ -54,16 +54,20 @@ public async Task<TransportResponse> GetRequest(TransportRequest transportReques
                     Headers = httpResult.Headers.ToDictionary(h => h.Key, h => h.Value),
                     RequestUrl = httpResult.RequestMessage?.RequestUri?.AbsolutePath
                 };
-                logger?.Debug($"Received http response from server with status code {httpResult.StatusCode}, content-length: {transportResponse.Content.Length} bytes, for url {transportRequest.RequestUrl}");
+                logger?.Debug($"HttpClient Service:Received http response from server with status code {httpResult.StatusCode}, content-length: {transportResponse.Content.Length} bytes, for url {transportRequest.RequestUrl}");
             }
-            catch (TaskCanceledException)
+            catch (TaskCanceledException taskCanceledException)
```

Comment:
and here Transport layer providing exception. So that inner exception can be inspected.